src/tls.rs

```diff
@@ -17,9 +17,20 @@ pub fn load_certs(path: &Path) -> std::io::Result<Vec<Certificate>> {
 }
 
 pub fn load_keys(path: &Path) -> std::io::Result<Vec<PrivateKey>> {
-    rsa_private_keys(&mut std::io::BufReader::new(std::fs::File::open(path)?))
-        .map_err(|_| std::io::Error::new(std::io::ErrorKind::InvalidInput, "invalid key"))
-        .map(|mut keys| keys.drain(..).map(PrivateKey).collect())
+    let mut rd = std::io::BufReader::new(std::fs::File::open(path)?);
+    let mut keys = Vec::<Vec<u8>>::new();
+
+    loop {
+        match read_one(&mut rd)? {
+            None => break,
+            Some(Item::RSAKey(key)) => keys.push(key),
+            Some(Item::PKCS8Key(key)) => keys.push(key),
+            Some(Item::ECKey(key)) => keys.push(key),
+            _ => {}
```

Comment:
Isn't this an infinite loop?

Reply:
I replaced the infinite loop with iter in commit 8f1f5ad.